src/PPrintEngine.ml

```diff
@@ -79,14 +79,17 @@ class buffer_output buffer = object
 end
 
 class formatter_output fmt = object
-  method char = Format.pp_print_char fmt
+  method char = function
+    | '\n' -> Format.pp_force_newline fmt ()
+    | ' '  -> Format.pp_print_space fmt ()
+    | c    -> Format.pp_print_char fmt c
+
   method substring str ofs len =
-    let str' =
+    Format.pp_print_string fmt (
```

Comment:
Is it intentional that you still use `pp_print_string` whereas you said you would use `pp_print_text`?

Reply:
(awaiting your answer on this point?)

Reply:
That was a mistake, sorry for that. It is now fixed, thanks :).